torch/nn/functional.py

```diff
@@ -742,3 +742,16 @@ def triplet_margin_loss(anchor, positive, negative, margin=1.0, p=2, eps=1e-6, s
     dist_hinge = torch.clamp(margin + d_p - d_n, min=0.0)
     loss = torch.mean(dist_hinge)
     return loss
+
+
+def normalize(input, p, dim, eps=1e-12):
+    r"""Performs l_p normalization of inputs over specified dimension.
+
+    Does:
+
+    .. math::
+        v = \frac{v}{\max(\lVert v \rVert_p, \epsilon)}
+
+    for each subtensor v over dimension dim of input.
```

Comment:
it might be worth adding that each `v` is viewed as a vector (if `input` is a 3D tensor, then the sub-matrices **wouldn't** be normalized w.r.t. a matrix norm).